macros/cross-adapter-modeling/sessionization/segment_web_sessions__initial.sql

```diff
@@ -0,0 +1,95 @@
+{% macro segment_web_sessions__initial() %}
+
+    {{ adapter_macro('segment.segment_web_sessions__initial') }}
+
+{% endmacro %}
+
+
+{% macro default__segment_web_sessions__initial() %}
+
+{{ config(
+    materialized = 'incremental',
+    sql_where = 'TRUE',
+    unique_key = 'session_id',
+    sort = 'session_start_tstamp',
+    dist = 'session_id'
+    )}}
+
+{% set partition_by = "partition by session_id" %}
+
+{% set window_clause = "
+    partition by session_id 
+    order by page_view_number 
+    rows between unbounded preceding and unbounded following
+    " %}
+    
+{% set first_values = {
+    'utm_source' : 'utm_source',
+    'utm_content' : 'utm_content',
+    'utm_medium' : 'utm_medium',
+    'utm_campaign' : 'utm_campaign',
+    'utm_term' : 'utm_term',
+    'mkw_id' : 'mkw_id',
+    'gclid' : 'gclid',
+    'page_url' : 'first_page_url',
+    'page_url_host' : 'first_page_url_host',
+    'page_url_path' : 'first_page_url_path',
+    'page_url_query' : 'first_page_url_query',
+    'referrer' : 'referrer',
+    'referrer_host' : 'referrer_host'
+    } %}
+    
+{% set last_values = {
+    'page_url' : 'last_page_url',
+    'page_url_host' : 'last_page_url_host',
+    'page_url_path' : 'last_page_url_path',
+    'page_url_query' : 'last_page_url_query'
+    } %}
+
+with pageviews as (
+
+    select * from {{ref('segment_web_page_views__sessionized')}}
+
+    {% if adapter.already_exists(this.schema, this.table) and not flags.FULL_REFRESH %}
+        where tstamp > (select dateadd(hour, -3, max(session_start_tstamp)) from {{ this }})
```

Comment:
I see we're accounting for the possibility of having split sessions by regularly re-calculating the past 3 hours. Should this be a `var`, or is there a reason why it's always 3?

Reply:
you're absolutely right--should be a variable. updated.